## Student: Kevin Rojas Bohorquez

### Exercise 1
Cargamos el dataset ***Choices.csv*** y visualizamos las primeras lineas:

In [4]:
choices <- read.table("Choices.csv", header = TRUE, sep=";")
choices[1:3,]

ID COUNTRY CHOICE_ID INFO MEASURE
1 1  6       1         A    IV     
2 2  2       1         A    IV     
3 3  2       1         A    IV

Para ver la dimensionalidad usamos la funcion ***dim***:

In [18]:
dim(choices)

[1] 36000     5

Para saber los nombres de las variables usamos la funcion ***names***:

In [19]:
names(choices)

[1] "ID"        "COUNTRY"   "CHOICE_ID" "INFO"      "MEASURE"

Y para saber las clases usamos la funcion ***sapply*** con la funcion ***class***:

In [22]:
sapply(choices,class)

ID   COUNTRY CHOICE_ID      INFO   MEASURE 
"integer" "integer" "integer"  "factor"  "factor"

Una mejor forma de ver todo lo requirido es usando la funcion ***str***:

In [26]:
str(choices)

'data.frame':	36000 obs. of  5 variables:
 $ ID       : int  1 2 3 4 5 6 7 8 9 10 ...
 $ COUNTRY  : int  6 2 2 2 2 2 2 2 2 2 ...
 $ CHOICE_ID: int  1 1 1 1 1 2 2 1 1 1 ...
 $ INFO     : Factor w/ 4 levels "A","B","C","D": 1 1 1 1 1 2 2 1 1 1 ...
 $ MEASURE  : Factor w/ 5 levels "I","II","III",..: 4 4 4 4 4 3 3 4 4 4 ...


Y otra forma, pero con datos estadisticos, es la funcion ***describe*** y se requiere cargas la libreria ***psych***:

In [27]:
library(psych)
describe(choices)

vars n     mean        sd           median trimmed     mad       min
ID        1    36000 1800.500000 1039.2448785 1800.5 1800.500000 1334.3400 1  
COUNTRY   2    36000    3.500000    1.7078488    3.5    3.500000    2.2239 1  
CHOICE_ID 3    36000    1.412472    0.4922861    1.0    1.390590    0.0000 1  
INFO*     4    36000    2.362667    1.0581974    2.0    2.328333    1.4826 1  
MEASURE*  5    36000    2.931667    1.3610224    3.0    2.914583    1.4826 1  
          max  range skew       kurtosis  se         
ID        3600 3599  0.00000000 -1.200100 5.477301438
COUNTRY      6    5  0.00000000 -1.268668 0.009001154
CHOICE_ID    2    1  0.35558730 -1.873610 0.002594576
INFO*        4    3  0.11548512 -1.219865 0.005577190
MEASURE*     5    4  0.08787589 -1.258404 0.007173218

### Exercise 2
Creamos la tabla de contingencia ***conTableIM*** con la funcion ***table***:

In [35]:
levels(choices$INFO)
levels(choices$MEASURE)
conTableIM<-table(choices$INFO,choices$MEASURE)
conTableIM

[1] "A" "B" "C" "D"

[1] "I"   "II"  "III" "IV"  "V"

   
       I   II  III   IV    V
  A 1892 2625 1708 2115 1414
  B 1360 2669 1485 2240 1923
  C 1677 2186 2093 2745 1627
  D 1507 1973  855  975  931

### Exercise 3
Obtenemos la distribucion de frecuencias marginales con la funcion ***addmargins***:

In [38]:
addmargins(conTableIM)

I    II   III  IV   V    Sum  
  A   1892 2625 1708 2115 1414  9754
  B   1360 2669 1485 2240 1923  9677
  C   1677 2186 2093 2745 1627 10328
  D   1507 1973  855  975  931  6241
  Sum 6436 9453 6141 8075 5895 36000

### Exercise 4
Usamos la funcion ***summary*** para obtener detalles de la tabla de la contigencia:

In [42]:
summary(conTableIM)

Number of cases in table: 36000 
Number of factors: 2 
Test for independence of all factors:
	Chisq = 860.7, df = 12, p-value = 1.602e-176

Como observamos tenemos un Chi cuadrado alto(***Chisq = 860.7***) y p-value despreciable (***p-value = 1.602e-176***), lo que indica que la hipotesis que la variable *INFO* y la variable *MEASURE* sean independientes es casi nula. Por ende las variables son dependientes. 